share/completions/loadkeys.fish

```diff
@@ -0,0 +1,26 @@
+function __fish_print_console_keymaps
+    # The path(s) used may differ by linux distribution, and are compiled into
+    # loadkeys, which doesn't provide a way to check which paths are searched
+    # or which keymaps it can find. localectl can list keymaps, but is part of
+    # systemd, which we shouldn't depend on.
+
+    for d in '/usr/share/kbd/keymaps' '/usr/share/keymaps' '/usr/lib/kbd/keymaps' '/lib/kbd/keymaps' '/usr/src/linux/drivers'
+        [ -d $d ] && find $d -type f | string replace -rf '.*/(.*)\.k?map(|\.gz|\.bz2|\.xz|\.zst)$' '$1'
```

Comment:
This can be simplified somewhat:

```fish
set -l dirs /usr/share/kbd/keymaps /usr/share/keymaps /usr/lib/kbd/keymaps /lib/kbd/keymaps /usr/src/linux/drivers
path filter -f $dirs/** | string replace -rf '.*/(.*)\.k?map(|\.gz|\.bz2|\.xz|\.zst)$' '$1'
```

No need to run `find` five times or test for the existence of the directories separately. You can just glob in them and keep the files.


---

Another thing: I wouldn't want to hardcode the compression extension here - I'd just look for a `\.k?map` and throw away any extension after that, so:

```fish
string replace -rf '.*/(.*)\.k?map(|\..*)$' '$1'
```

Now, technically this could be mis-matching something if there was ever a file with a different extension there - `foo.map.this_is_not_a_map`. However, I would call that *less* likely than someone using a compression algorithm you don't know here.

Reply:
I doubt supported compression schemes are going to change soon, and there are a bunch of weird extensions in `/usr/src/linux`, though I guess even if a false positive triggering file was added, it wouldn't be that harmful; I'll apply that.

Also sorry, I didn't know `path` had special behaviour for failed wildcard matches, so I didn't want to use that, as e.g. Ubuntu's `loadkeys` comes without any keymaps. I'll apply that as well.